content/articles/continuous-integration-and-deployment-of-pipelines-with-flask-docker-and-github-actions/index.md

```diff
@@ -0,0 +1,291 @@
+
+Now the `jobs` are what to do at every step. In this, we want to deploy on the `ubuntu-latest` machine.  Deploying the code requires the steps as defined. This step will run by installing some dependencies described, and on successful build, you have all the steps marked. Something like the image below;
+
+![Successful Build](/continuous-integration-and-deployment-of-pipelines-with-flask-docker-and-github-actions/publish.png)
+
+### Docker images and containers basics
+In the past, the default way to deploy applications was to spin up a `virtual machine`. This copies the necessary binaries' artifact and the execution in the background. 
+
+In essence, a virtual machine is composed of a host operating system with the sets of default packages installed. The application will use the operating system files and resources to run successfully.
+
+Additionally, the virtual machine provides the benefits of infrastructure through the `hypervisor`. This hypervisor allows many virtual machines to run and is also used to create them on the operating system.
+
+However, the trade-off behind is the replication of the operating system. The more virtual machines run, the higher the space consumed in the host operating system. To optimize the backlog of this concept, the industry introduced the containerization of applications.
+
+Containers are the lightweight and reliable technique to deliver a product to consumers while having better use of available resources.
+
+Instead of having multiple virtual machines, we can have one hosting system that runs multiple containers. The processes in the containers are isolated and they have access to the file system with packages on the host operating system.
+
+The creation and execution of the containers are delegated to a container management tool such as `Docker`. This OS-level virtualization unlocks the benefits of running multiple applications by using the containers.
+
+In the following section, we will look at how to containerize applications using Docker and the `Dockerfile` dependency. 
+
+Create a `Dockerfile` inside the project folder and add the Docker commands below.
+
+```dockerfile
+FROM python:3.8
+WORKDIR /app
+COPY . .
+RUN pip install -r requirements.txt
+ENV PORT=80
+```
+
+Furthermore, We have to build the Docker image and then push it to the Docker hub for storage. But before that, create a new repository in Docker hub that will store your Docker image from the build actions.
+
+You will add your Docker hub account details to the GitHub secrets. You can do that by going to your GitHub account, clicking on the `settings` button, and tapping the `secrets`.
+
+Add the following to the name and values:
+
+```yaml
+DOCKER_USERNAME: <your-username>
+DOCKER_PASSWORD: <your-password>
+DOCKER_HUB: <your-username>/<your-repository>
```

Comment:
@Horlawhumy-dev
 These keys don't match the ones in your yaml file. Please update this.

Reply:
Updated sir